core/src/main/scala/org/apache/spark/api/java/JavaSparkContext.scala

```diff
@@ -670,6 +670,19 @@ class JavaSparkContext(val sc: SparkContext)
   }
 
   /**
+   * Add a file to be downloaded with this Spark job on every node.
+   * The `path` passed can be either a local file, a file in HDFS (or other Hadoop-supported
+   * filesystems), or an HTTP, HTTPS or FTP URI.  To access the file in Spark jobs,
+   * use `SparkFiles.get(fileName)` to find its download location.
+   *
+   * A directory can be given if the recursive option is set to true. Currently directories are only
+   * supported for Hadoop-supported filesystems.
+   */
+  def addFile(path: String, recursive: Boolean): Unit = {
```

Comment:
@srowen After investigating the code, I found it's not very straightforward to clean up the interfaces at ```JavaSparkContext```, since they were called by Python and R. For Python side, we can use ```_jsc.sc()``` in some cases, but it's messy if we use both ```JavaSparkContext``` and ```JavaSparkContext.sc``` at R side. So I think we should leave it as it is, or any other suggestion? Thanks.